### ISSUE TITLE
Add unknown case to FundingType

### ISSUE THREAD

**Issue by marcgeeklaberge** [stripe/stripe-ios#1960]
https://github.com/stripe/stripe-ios/blob/294dac6bb5b60b236c0f88d835ed768d7c25ef4d/StripeApplePay/StripeApplePay/Source/PaymentsCore/API/Models/Token.swift#L115

This needs to be changed by `case other, unknown` otherwise if the funding type is "unknown", the SDK is unable to decode properly the JSON response.

Example of response that can't be decode without this change:
```
{
  "id": ["tok_1KyElCRii06EefEmOTkeCqd8"](https://dashboard.stripe.com/test/logs?object=tok_1KyElCRii06EefEmOTkeCqd8),
  "object": "token",
  "card": {
    "id": "card_1KyElCRii06EefEmX1PAzf7J",
    "object": "card",
    "address_city": "Cerkvenjak",
    "address_country": "SI",
    "address_line1": "Trg revolucije 16",
    "address_line1_check": "unchecked",
    "address_line2": null,
    "address_state": null,
    "address_zip": "2236",
    "address_zip_check": "unchecked",
    "brand": "MasterCard",
    "country": "US",
    "cvc_check": null,
    "dynamic_last4": "0807",
    "exp_month": 9,
    "exp_year": 2022,
    "funding": "unknown",
    "last4": "1505",
    "name": "Marc Laberge",
    "tokenization_method": "apple_pay",
    "use": "unknown"
  },
  "client_ip": "209.226.99.87",
  "created": 1652272522,
  "livemode": false,
  "type": "card",
  "used": false
}
```

**Comment by marcgeeklaberge** [comment]
According to the following doc, the possible cases should be: `credit`, `debit`, `prepaid`, or `unknown`
https://stripe.com/docs/api/tokens/object#token_object-card-funding

**Comment by yuki-stripe** [comment]
Thank you @marcgeeklaberge for reporting this!  This will be fixed in the next release this coming Monday.

**Comment by marcgeeklaberge** [comment]
> Thank you @marcgeeklaberge for reporting this!  This will be fixed in the next release this coming Monday.

My pleasure, thanks you for releasing it that fast 🙏

Can you also make sure that that stripe react native is aligned with this version that will contain this fix please. We had to bridge on the iOS SDK since the react native version was too limitative for now, but we use it for credit card and possibly Alipay and we need to make sure that the Pod versions match 😊

Thank you 🙌

**Comment by vecsdom** [comment]
> https://github.com/stripe/stripe-ios/blob/294dac6bb5b60b236c0f88d835ed768d7c25ef4d/StripeApplePay/StripeApplePay/Source/PaymentsCore/API/Models/Token.swift#L115
> 
> 
> 
> This needs to be changed by `case other, unknown` otherwise if the funding type is "unknown", the SDK is unable to decode properly the JSON response.
> 
> 
> 
> Example of response that can't be decode without this change:
> 
> ```
> 
> {
> 
>   "id": ["tok_1KyElCRii06EefEmOTkeCqd8"](https://dashboard.stripe.com/test/logs?object=tok_1KyElCRii06EefEmOTkeCqd8),
> 
>   "object": "token",
> 
>   "card": {
> 
>     "id": "card_1KyElCRii06EefEmX1PAzf7J",
> 
>     "object": "card",
> 
>     "address_city": "Cerkvenjak",
> 
>     "address_country": "SI",
> 
>     "address_line1": "Trg revolucije 16",
> 
>     "address_line1_check": "unchecked",
> 
>     "address_line2": null,
> 
>     "address_state": null,
> 
>     "address_zip": "2236",
> 
>     "address_zip_check": "unchecked",
> 
>     "brand": "MasterCard",
> 
>     "country": "US",
> 
>     "cvc_check": null,
> 
>     "dynamic_last4": "0807",
> 
>     "exp_month": 9,
> 
>     "exp_year": 2022,
> 
>     "funding": "unknown",
> 
>     "last4": "1505",
> 
>     "name": "Marc Laberge",
> 
>     "tokenization_method": "apple_pay",
> 
>     "use": "unknown"
> 
>   },
> 
>   "client_ip": "209.226.99.87",
> 
>   "created": 1652272522,
> 
>   "livemode": true,
> 
>   "type": "card",
> 
>   "used": true
> 
> }
> 
> ```

**Comment by davidme-stripe** [comment]
We've released this fix as [version 22.3.1](https://github.com/stripe/stripe-ios/releases/tag/22.3.1). We also changed the way enum deserialization works, so an unknown enum value will no longer cause the entire object to fail deserialization.
